src/commands/schema/pull.mjs

```diff
@@ -52,51 +53,80 @@ function logDiff({ argv, adds, overwrites, deletes }) {
   }
 }
 
+/**
+ * @param {object} argv
+ * @returns {"active" | "staged" | undefined} The source to pull from
+ */
+function parsePullSource(argv) {
+  if (argv.active && argv.staged) {
+    throw new ValidationError("Cannot specify both --active and --staged.");
+  } else if (argv.active) {
+    return "active";
+  } else if (argv.staged) {
+    return "staged";
+  }
+
+  return undefined;
+}
+
 async function doPull(argv) {
   const logger = container.resolve("logger");
   const confirm = container.resolve("confirm");
   const makeFaunaRequest = container.resolve("makeFaunaRequest");
   const secret = await getSecret();
 
+  const argvSource = parsePullSource(argv);
+  const filesParams =
+    argvSource === "staged"
+      ? new URLSearchParams({ staged: "true" })
+      : undefined;
+
   // fetch the list of remote FSL files
   const filesResponse = await makeFaunaRequest({
     argv,
     path: "/schema/1/files",
+    params: filesParams,
     method: "GET",
     secret,
   });
 
+  const version = filesResponse.version;
+
   // sort for consistent order (it's nice for tests)
   const filenames = filesResponse.files
     .map((file) => file.filename)
     .filter((name) => name.endsWith(".fsl"))
     .sort();
 
-  // check if there's a staged schema
+  // Get the staged schema status
+  /** @type {{ status: "none" | "pending" | "ready" | "failed" }} */
   const statusResponse = await makeFaunaRequest({
     argv,
     path: "/schema/1/staged/status",
-    params: new URLSearchParams({ version: filesResponse.version }),
+    params: new URLSearchParams({ version: version }),
     method: "GET",
     secret,
   });
 
-  // if there's a staged schema, cannot use the --active flag.
-  // getting active FSL while staged FSL exists is not yet
-  // implemented at the service level.
-  if (statusResponse.status !== "none" && argv.active) {
+  // If the source was not specified by the caller, determine it based on the status
+  if (argvSource === "staged" && statusResponse.status === "none") {
     throw new CommandError(
-      "There is a staged schema change. Remove the --active flag to pull it.",
+      "There is no staged schema to pull. Remove --staged or specify --active instead.",
     );
-  } else if (statusResponse.status === "none" && !argv.active) {
-    throw new CommandError("There are no staged schema changes to pull.");
   }
+  const source =
+    (argvSource ?? statusResponse.status === "none") ? "active" : "staged";
+
+  logger.debug(
+    `Pulling remote ${source} schema, version '${version}'.`,
+    "schema pull",
```

Comment:
@echo-bravo-yahoo do we have a preference of these being one word to simplify passing them?

Reply:
yeah i don't think it would work as expected unless it was a single or hyphenated word